avocado/utils/dmesg.py

```diff
@@ -115,3 +121,23 @@ def collect_errors_by_level(output_file=None, level_check=5):
             err += " Please check  dmesg log in debug log."
             LOGGER.debug(dmsg_log)
         raise DmesgError("Test is failed {}".format(err))
+
+
+def skip_dmesg_messages(dmesg_stdout, skip_messages):
+    """Remove some messages form dmesg output.
+
+    This method will remove some lines form dmesg output if some
+    strings are present. Returning the filter output.
+
+    :dmesg_stdout: dmesg messages from which want to filter
+                   it should be decoded string.
+    :type dmesg_stdout: string
+    :skip_messages: list of message for remove
+    :type skip_messages: list
+    """
+    for dmesg in skip_messages:
+        for line in dmesg_stdout.splitlines():
+            if dmesg in line:
+                dmesg_stdout = dmesg_stdout.replace(line, "")
+    filter_string = [line for line in dmesg_stdout.split('\n') if line.strip()]
```

Comment:
I don't understand why using this approach instead of a much simpler version. Do you have any reason for that?

Also, this is assuming that new lines boundaries are always going to be a `\n`. Today we have more or less 13 boundaries line separators (`\r`, `\n`, `\r\n`, `\v`, `\x0b`, `\u2028`, etc..) depending on your system. That is one of the reasons that you are using `\s` and `.splitlines()`, so they will take care of this for you.

Returning a `\n` as line separator it is OK, but unless I'm missing something, assuming that all separators are going to be `\n` IMO it is not good.

Reply:
i don't have any reason,i just simplify that approach it might work out. i am completely fine with the approach what you have suggested will go with that.